File: src/main/java/com/uwetrottmann/trakt5/entities/CheckinError.java
package com.uwetrottmann.trakt5.entities;

import retrofit2.Response;

import java.util.Date;

/**
 * Type to use for parsing check in error response (call {@link com.uwetrottmann.trakt5.TraktV2#checkForCheckinError(Response)}
 * with this class) if you get a 409 HTTP status code when checking in.
 */
public class CheckinError {

    /** Timestamp which is when the user can check in again. */
    public Date expires_at;

}

File: src/main/kotlin/com/redissi/trakt/entities/CheckinError.kt
package com.redissi.trakt.entities

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.time.OffsetDateTime

/**
 * Type to use for parsing check in error response (call [com.redissi.trakt.Trakt.checkForCheckinError]
 * with this class) if you get a 409 HTTP status code when checking in.
 */
@JsonClass(generateAdapter = true)
class CheckinError(
    /** Timestamp which is when the user can check in again.  */
    @Json(name = "expires_at")
    val expiresAt: OffsetDateTime? = null
)